autospec/tarball.py

```diff
@@ -143,24 +143,55 @@ def build_unzip(zip_path):
     ***snip***
     and this function gets the 'prefix-dir' portion from the start of the unzip -l output.
     """
-    prefix = ""
+    prefix = None
     contents = subprocess.check_output(["unzip", "-q", "-l", zip_path], universal_newlines=True)
     lines = contents.splitlines() if contents else []
     # looking for directory prefix in unzip output as it may differ from default
-    if len(lines) > 2:
-        # In some cases there is a hash line so we detect this based on the header
-        # separator character '-'
-        prefix_line = 3 if len(lines) > 3 and lines[2][0] == '-' else 2
-        prefix = lines[prefix_line].split("/")[0].split()[-1]
-    else:
+
+    # First, find the end of the info header
+    # It starts with -----
+    while len(lines):
+        line = lines.pop(0)
+        if line[0] == '-':
+            break
+
+    # Similarly, strip the footer
+    while len(lines):
+        line = lines.pop()
+        if line[0] == '-':
+            break
+
+    # Are there any files left?
+    if len(lines) < 1:
         print_fatal("Zip file doesn't appear to have any content")
         exit(1)
 
+    # Look for a common directory prefix
+    for line in lines:
+        if line[0] == '-':
```

Comment:
Under what condition does it hit this? I expect the header and footer being removed would cause no lines with '-' as the first character.

Reply:
Whoops, relic to be deleted.